src/main/java/picard/util/IntervalList/IntervalListScatter.java

```diff
@@ -0,0 +1,108 @@
+package picard.util.IntervalList;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+
+import java.util.ArrayDeque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class IntervalListScatter implements Iterable<IntervalList> {
+
+    private final IntervalListScatterer scatterer;
+    private final IntervalList intervals;
+    private final int scatterCount;
+
+    public IntervalListScatter(final IntervalListScatterer scatterer, final IntervalList intervals, final int scatterCount) {
+        if (scatterCount < 1) {
+            throw new IllegalArgumentException("scatterCount < 1");
+        }
+        this.scatterer = scatterer;
+        this.intervals = scatterer.preprocessIntervalList(intervals);
+        this.scatterCount = scatterCount;
+    }
+
+    @Override
+    public Iterator<IntervalList> iterator() {
+        return new ScatterState(scatterer, intervals, scatterCount);
+    }
+
+    class ScatterState implements Iterator<IntervalList> {
+
+        private final int scatterCount;
+        private final int idealSplitWeight;
+        private final IntervalListScatterer scatterer;
+        private final SAMFileHeader header;
+
+        // Use a DeQueue since algo will be adding and removing elements from head.
+        final ArrayDeque<Interval> intervalQueue;
+        long weightRemaining;
+        long intervalsReturned = 0;
+
+        public ScatterState(final IntervalListScatterer scatterer, final IntervalList inputIntervals, final int scatterCount) {
+            this.scatterCount = scatterCount;
+            final IntervalList processedIntervals = scatterer.preprocessIntervalList(inputIntervals);
+            this.idealSplitWeight = scatterer.deduceIdealSplitWeight(processedIntervals, scatterCount);
+            this.intervalQueue = new ArrayDeque<>(processedIntervals.getIntervals());
+            this.scatterer = scatterer;
+            this.header = processedIntervals.getHeader();
+            this.weightRemaining = scatterer.listWeight(processedIntervals);
+        }
+
+        @Override
+        public boolean hasNext() {
+            return !intervalQueue.isEmpty();
+        }
+
+        @Override
+        public IntervalList next() {
+            intervalsReturned++;
```

Comment:
This increment will always happen, even if the exception is thrown because the user requested `next()` of the empty iterator.

Reply:
if this throws, who cares if intervalsReturned is off by one?